compiler/src/lib.rs

```diff
@@ -2,414 +2,25 @@
 
 #![deny(clippy::print_stdout)]
 
-use std::ffi::OsStr;
-use std::fs;
 use std::path::Path;
-use std::path::PathBuf;
-use std::time::Instant;
 
-use analysis::analyze;
-use analysis::convert_analyzed_to_pil_constraints;
 use ast::analyzed::Analyzed;
-use ast::DiffMonitor;
 
+pub mod pipeline;
+pub mod test_util;
 pub mod util;
-mod verify;
+pub mod verify;
 
-use ast::asm_analysis::AnalysisASMFile;
 pub use backend::{BackendType, Proof};
 use executor::witgen::QueryCallback;
 use itertools::Itertools;
-pub use verify::{
-    verify, verify_asm_string, write_commits_to_fs, write_constants_to_fs, write_constraints_to_fs,
-};
 
-use ast::parsed::PILFile;
-use executor::constant_evaluator;
 use number::FieldElement;
 
-pub fn no_callback<T>() -> Option<fn(&str) -> Option<T>> {
-    None
-}
-
-/// Compiles a .pil or .asm file and runs witness generation.
-/// If the file ends in .asm, converts it to .pil first.
-/// Returns the compilation result if any compilation took place.
-pub fn compile_pil_or_asm<T: FieldElement>(
-    file_name: &str,
-    inputs: Vec<T>,
-    output_dir: &Path,
-    force_overwrite: bool,
-    prove_with: Option<BackendType>,
-    external_witness_values: Vec<(&str, Vec<T>)>,
-) -> Result<Option<CompilationResult<T>>, Vec<String>> {
-    if file_name.ends_with(".asm") {
-        compile_asm(
-            file_name,
-            inputs,
-            output_dir,
-            force_overwrite,
-            prove_with,
-            external_witness_values,
-        )
-    } else {
-        Ok(Some(compile_pil(
-            Path::new(file_name),
-            output_dir,
-            inputs_to_query_callback(inputs),
-            prove_with,
-            external_witness_values,
-        )))
-    }
-}
-
 pub fn analyze_pil<T: FieldElement>(pil_file: &Path) -> Analyzed<T> {
     pil_analyzer::analyze(pil_file)
 }
 
```

Comment:
My comment about the "missing" API entry is related to this function that was leftover here.